runelite-client/src/main/java/net/runelite/client/plugins/woodcutting/WoodcuttingTimersOverlay.java

```diff
@@ -0,0 +1,135 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.woodcutting;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.time.Instant;
+import java.util.Iterator;
+import java.util.List;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.Perspective;
+import net.runelite.api.Point;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.ProgressPieComponent;
+
+class WoodcuttingTimersOverlay extends Overlay
+{
+	// Range of Trees respawn time
+	static final int TREE_MAX_RESPAWN_TIME = 60;
+	private static final int TREE_MIN_RESPAWN_TIME = 30;
+	private static final float TREE_RANDOM_PERCENT_THRESHOLD = (float) TREE_MIN_RESPAWN_TIME / TREE_MAX_RESPAWN_TIME;
+	private static final Color DARK_GREEN = new Color(0, 100, 0);
+
+	private final Client client;
+	private final WoodcuttingPlugin plugin;
+
+	@Inject
+	private WoodcuttingTimersOverlay(Client client, WoodcuttingPlugin plugin)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		this.plugin = plugin;
+		this.client = client;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		List<TreeRespawn> respawns = plugin.getRespawns();
+		if (respawns.isEmpty())
+		{
+			return null;
+		}
+
+		Instant now = Instant.now();
+		for (Iterator<TreeRespawn> it = respawns.iterator(); it.hasNext();)
+		{
+			Color pieFillColor = Color.YELLOW;
+			Color pieBorderColor = Color.ORANGE;
+			TreeRespawn treeRespawn = it.next();
+			float percent = (now.toEpochMilli() - treeRespawn.getStartTime().toEpochMilli()) / (float) treeRespawn.getRespawnTime();
+			WorldPoint worldPoint = treeRespawn.getWorldPoint();
+			LocalPoint loc = LocalPoint.fromWorld(client, worldPoint);
+
+			// This is used to get the X and Y for centre location to work
+			LocalPoint tempLoc = LocalPoint.fromWorld(client, worldPoint.getX(), worldPoint.getY());
+			LocalPoint centreLoc = new LocalPoint(
+				tempLoc.getX() + Perspective.LOCAL_TILE_SIZE * (client.getPlane() - 1) / 2,
+				tempLoc.getY() + Perspective.LOCAL_TILE_SIZE * (client.getPlane() - 1) / 2);
+			if (loc == null || percent > 1.0f)
+			{
+				it.remove();
```

Comment:
Why is the overlay responsible for removing trees which have respawned? That logic should be contained in the client, and the overlay should be responsible only for drawing the respawn timers.

Reply:
This is how it's done with the mining timers overlay, which is the reason I've done it here